File: WeatherApp 1/WeatherApp/Weather.m
//
//  Weather.m
//  WeatherApp
//
//  Created by projas on 2/9/16.
//  Copyright © 2016 projas. All rights reserved.
//

#import "Weather.h"

@implementation Weather

- (instancetype)initWithTemperature:(NSNumber *)temperature{
    self = [super init];
    if (self) {
        self.temperature = @(round([temperature floatValue]));
    }
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dictionary{
    self = [super init];
    if (self) {
        self.date = [NSDate dateWithTimeIntervalSince1970:[dictionary[@"dt"] floatValue]];
        self.temperature = dictionary[@"temp"][@"day"];
        self.minTemperature = @(round([dictionary[@"temp"][@"min"] floatValue]));
        self.maxTemperature = @(round([dictionary[@"temp"][@"max"] floatValue]));
        
    }
    return self;
}

@end

File: WeatherApp-Swift/WeatherApp-Swift/Weather.swift
//
//  Weather.swift
//  WeatherApp-Swift
//
//  Created by projas on 3/28/16.
//  Copyright © 2016 projas. All rights reserved.
//

import UIKit
import ObjectMapper

class Weather: Mappable {
    
    var temperature: Double = 0.0
    var maxTemperature: Double = 0.0
    var minTemperature: Double = 0.0
    var date: NSDate?
    
    init(WithTemperature temperature: Double){
        self.temperature = temperature
    }
    
    required init?(_ map: Map) {
        
    }
    
    func mapping(map: Map) {
        temperature         <- map["temp.day"]
        maxTemperature      <- map["temp.max"]
        minTemperature      <- map["temp.min"]
        date                <- (map["dt"], DateTransform())
    }

}
